src/IntDict.elm

```diff
@@ -707,17 +726,18 @@ uniteWith merger l r =
     let
         mergeWith key left right =
             case ( left, right ) of
-                ( Just l, Just r ) ->
-                    Just (merger key l r)
+                ( Just l2, Just r2 ) ->
+                    Just (merger key l2 r2)
 
-                ( Just l, _ ) ->
+                ( Just _, _ ) ->
                     left
 
-                ( _, Just r ) ->
+                ( _, Just _ ) ->
                     right
 
                 ( Nothing, Nothing ) ->
-                    Debug.crash "IntDict.uniteWith: mergeWith was called with 2 Nothings. This is a bug in the implementation, please file a bug report!"
+                    -- This is a bug in the implementation, please file a bug report!
+                    Nothing
```

Comment:
I find this very unfortunate. Isn't there some alternative to `Debug.crash`? I'd rather crash than possibly some incorrect result because of a broken invariant. Or at least convert this to a `Debug.log` or `Debug.todo` message.

Reply:
https://alpha.elm-lang.org/packages/elm/core/latest/Debug#todo

Explicitly says that Debug.todo is forbidden in packages.

Reply:
In that case, this is the death to die.

Reply:
I noticed that in [Evan's code](https://github.com/elm/browser/blob/08992348335f62da73afdeeb025175a528485363/src/Debugger/History.elm#L203) he replaced `Debug.crash` with inifinite loops. You might want to do something similar as it seems crazy to allow bad data to sneak through.

Reply:
Wow, nontermination is a lot more painful to detect and is semantically equivalent to just crashing. This all still sounds like a bad joke to me, but will do.

Reply:
After considering the relevant code again, returning `Nothing` seems a fine fall back and will not lead to bad data.